docs/network-spec/connection-manager.tex

```diff
@@ -273,7 +356,212 @@ \section{Components}
+  connection -- either fresh or pre-existing. To complete the promotion it will
+  start the client side of particular mini-protocols on the connection.
+
+\item[Release a connection.] The governor also decides when to demote a peer to
+  cold. As part of the demotion the client side mini-protocols are terminated.
+  The connection is then no longer needed by the governor and is released.
+\end{description}
+
+\begin{designalternative}
+The code currently uses the names |requestOutboundConnection| and
+|unregisterOutboundConnection|. The names `request' and 'unregister' are not a
+pair of names that fit well together. The dual of `unregister' would be
+'register'. A more consistent pair of names might be `acquire' and `release',
+and hence API names |acquireOutboundConnection| and |releaseOutboundConnection|.
+\end{designalternative}
+
+It is worth noting again that the P2P governor does not require exclusive
+access to the TCP bearer. It has no special TCP-specific needs in the setup or
+shutdown.
+It needs access to the multiplexer to be able to run a set of
+mini-protocols in one direction. So in a sense it needs exclusive access to
+`half' a multiplexer for a connection, but it does not need coordinate with or
+even to be aware of any use of the other `half' of the multiplexer. It is this
+separation of concerns that enables a modular design and implementation.
+
+
+\section{Inbound side: the server}
+
+The inbound side has a less complex task than the P2P governor, but its
+interactions with the connection manager are slightly more complicated.
+
+The inbound side is split into two components: the server and inbound governor.
+
+The server is responsible for accepting new TCP connections on the listening
+socket. It is responsible for not exceeding resource limits by accepting too
+many new connections. It is also responsible for a little bit of DoS protection:
+limiting the rate of accepting new connections.
+
+The server component is much simpler than in most network server applications
+because it does not need to manage the connection resources once created. The
+server hands new connections over to the connection manager as soon as they are
+accepted. The server's responsibilities end there. The server needs only two
+interactions with the the connection manager.
+\begin{description}
+\item[Query number of connections] The server component needs to query the
+  connection manager to find the current number of connections. It uses this
+  information to decide if any new connections can be accepted or if we are
+  at the resource limits. Below the hard limits, the current number can be used
+  as part of rate limiting decisions.
+
+\item[Hand over a new connection] Once the server component has successfully
+  accepted a new connection, it needs to hand over responsibility for it to
+  the connection manager.
+  \todo[inline]{be clearer in the responsiblities and the conditions that should hold}
```

Comment:
I don't think there's much to say here besides that only connections that are allowed by rate limiting make it through